README.md

```diff
@@ -75,3 +81,163 @@ rules:
+You will see in [TLS Certificates](#tls-certificates) section below that this label is also used to inject the CA bundle into the webhook configuration.
+
+ > *Note*: if you plan running validations on `DeploymentSet` or `ReplicaSet` objects, you need to make sure that the `rollout-operator` has the privileges to list and get those objects.
+
+##### Handling errors
+
+The webhook is conservative and allows changed whenever an error occurs:
+- When parent object can't be retrieved from the API.
+- When the validation request can't be decoded or includes an unsupported type.
+
+##### Special cases
+
+Changing the replicas number to `null` (or from `null`) is allowed.
+
+### TLS Certificates
+
+Note that `ValidatingAdmissionWebhook` require a TLS connection, so the HTTPS server should either use a certificate signed by a well-known CA or a self-signed certificate.
+You can either [issue a Certificate Signing Request](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/) or use an existing approach for issuing self-signed certificates, like [cert-manager](https://cert-manager.io/docs/configuration/selfsigned/).
+You can set the options `-server-tls.cert-file` and `-server-tls.key-file` to point to the certificate and key files respectively.
+
+For convenience, `rollout-operator` offers a self-signed certificates generator that is enabled by default.
+This generator will generate a self-signed certificate and store it in a secret specified by the flag `-server-tls.self-signed-cert.secret-name`.
+The certificate is stored in a secret in order to reuse it across restarts of the `rollout-operator`.
+
+`rollout-operator` will list all the `ValidatingWebhookConfiguration` objects in the cluster that are labeled with `grafana.com/inject-rollout-operator-ca: true` and `grafana.com/namespace: <value of -kubernetes-namespace>` and will inject the CA certificate in the `caBundle` field of the webhook configuration. 
+This mechanism can be disabled by setting `-webhooks.update-ca-bundle=false`.
+
+This signing and injecting is performed at service startup once, so you would need to restart `rollout-operator` if you want to inject the CA certificate in a new `ValidatingWebhookConfiguration` object.
+
+In order to perform the self-signed certificate generation, `rollout-operator` needs a `Role` that would allow it to list and update the secrets, as well as a `ClusterRole` that would allow listing and patching the `ValidationWebhookConfigurations`. An example of those could be:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: rollout-operator-webhook-role
+  namespace: default
+rules:
+  - apiGroups: [""]
+    resources: [secrets]
+    verbs: [create, update, get]
```

Comment:
Good idea, but not trivial, because adding a `resourceNames` here implies that we need the secret to be created beforehand. Did that in https://github.com/grafana/rollout-operator/pull/25/commits/5ef7b42903dd745fe02bb9e2f6550863d9a3468a